metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/DirectSqlTable.java

```diff
@@ -169,56 +176,93 @@ public boolean exists(final QualifiedName name) {
     }
 
     /**
-     * Locks the iceberg table for update so that no other request can modify the table.
-     * @param tableId table internal id
-     * @param tableName table name
+     *  Locks and updates the iceberg table for update so that no other request can modify the table at the same time.
+     *  1. Gets the table parameters and locks the requested records. If lock cannot be attained,
+     *  the request to update fails
+     *  2. Validates the metadata location
+     *  3. If validated, updates the table parameters.
+     * @param tableInfo table info
      */
     @Transactional(propagation = Propagation.REQUIRES_NEW)
-    public void lockIcebergTable(final Long tableId, final QualifiedName tableName) {
-        String tableType = null;
+    public void updateIcebergTable(final TableInfo tableInfo) {
+        final QualifiedName tableName = tableInfo.getName();
+        final Long tableId = getTableId(tableName);
+        Map<String, String> existingTableMetadata = Maps.newHashMap();
+        log.debug("Lock Iceberg table {}", tableName);
         try {
-            tableType = jdbcTemplate.queryForObject(SQL.TABLE_PARAM_LOCK,
-                new SqlParameterValue[] {new SqlParameterValue(Types.BIGINT, tableId),
-                    new SqlParameterValue(Types.VARCHAR, PARAM_TABLE_TYPE), }, String.class);
-        } catch (EmptyResultDataAccessException ignored) { }
-        if (tableType == null || !ICEBERG_TABLE_TYPE.equalsIgnoreCase(tableType)) {
-            final String message = String.format("Originally table %s is not of type iceberg", tableName);
-            log.info(message);
+            existingTableMetadata = jdbcTemplate.query(SQL.TABLE_PARAMS_LOCK,
+                new SqlParameterValue[]{new SqlParameterValue(Types.BIGINT, tableId)}, rs -> {
```

Comment:
Now we are locking all the rows of this table parameters.

Reply:
Yes.